tests/integration/test_charm.py

```diff
@@ -118,15 +128,60 @@ async def test_settings_are_correct(ops_test: OpsTest, series: str):
     assert settings["maximum_lag_on_failover"] == 1048576
 
 
-def build_application_name(series: str) -> str:
-    """Return a composite application name combining application name and series."""
-    return f"{APP_NAME}-{series}"
+@pytest.mark.parametrize("series", SERIES)
+async def test_scale_down_and_up(ops_test: OpsTest, series: str):
+    """Test data is replicated to new units after a scale up."""
+    # Set a composite application name in order to test in more than one series at the same time.
+    application_name = build_application_name(series)
+
+    # Ensure the initial number of units in the application.
+    initial_scale = len(UNIT_IDS)
+    await scale_application(ops_test, application_name, initial_scale)
+
+    # Scale down the application.
+    await scale_application(ops_test, application_name, initial_scale - 1)
```

Comment:
if possible include the leader unit and the primary unit in this scale down, then you can ignore my comments

-  "add checks for when the primary gets removed that the configuration is accurate (ie the juju unit ips match the PG ips)"
- "add a test for removing the juju leader"

Reply:
Added on https://github.com/canonical/postgresql-operator/pull/14/commits/6fe523270b2b5a4424f9ed91c7038de88e66812c (and also on https://github.com/canonical/postgresql-operator/pull/14/commits/b23967277d37d7d2291c48fe0653f71954b09f91, that contains additional fixes).